### PR TITLE
BlockFetcher/BlockUploader: fix and add additional attributes

### LINKED ISSUE #3631: Improve NeoFS BlockFetcher/BlockUploader object attributes
## Is your feature request related to a problem? Please describe.
Existing NeoFS containers that store Neo chains have two problems:
1. Object attributes start from lowercase letters: `block:0`, `index:0`.
2. Both block and index file objects don't have network magic included into attributes, and hence it's impossible to learn the network of the block or index file without additional external knowledge like this:
```
neo-testnet container: HmP48JvCcmiEbJemG7jFNEg2qkdCTx4thD3EwWUi288Q
neo-mainnet container: 2W2c7qHqicZJJdRG134k7eusZqidkk3KBDNgeLL45nn2
fschain testnet 4SJK36y4JGZJ5rjXsGWsJ3yiWrcvEANXBBHThQ6eErsq
fschain mainnet 5oAvDsnVcpGEf8KM4nxNB52wqZASjbr4CSszbW9iseCk
```

## Describe the solution you'd like
Reuploading currently used containers doesn't make sense, but for https://github.com/neo-project/neo/issues/3463 it should be done:
1. Capitalize block objects and index file objects attributes.
2. Add network magic attribute to both block and index file objects.

## Describe alternatives you've considered
Leave it as is, it also works.

### PR DESCRIPTION
**Pull request by AliceInHunterland**
Close #3631

### CODE REVIEW

File: cli/util/uploader.go [hunk 1]
```diff
@@ -83,6 +83,11 @@ func uploadBin(ctx *cli.Context) error {
 		return cli.Exit(fmt.Sprintf("failed to create RPC client: %v", err), 1)
 	}
 
+	vrs, err := rpc.GetVersion()
```
Comment (AnnaShaleva) [inline cli/util/uploader.go:86]: Just `v, err := ...`

File: cli/util/uploader.go [hunk 2]
```diff
@@ -83,6 +83,11 @@ func uploadBin(ctx *cli.Context) error {
...
+	if err != nil {
+		return cli.Exit(fmt.Sprintf("failed to get version from RPC: %v", err), 1)
+	}
+	magic := vrs.Protocol.Network.String()
```
Comment (AnnaShaleva) [inline cli/util/uploader.go:90]: It must be a number, not a "privnet" or any other string.

**Review by AnnaShaleva — DISMISSED** [review]

File: cli/util/uploader.go [hunk 3]
```diff
@@ -158,10 +163,11 @@ func uploadBin(ctx *cli.Context) error {
...
 						*object.NewAttribute(attr, strconv.Itoa(int(blk.Index))),
-						*object.NewAttribute("primary", strconv.Itoa(int(blk.PrimaryIndex))),
-						*object.NewAttribute("hash", blk.Hash().StringLE()),
-						*object.NewAttribute("prevHash", blk.PrevHash.StringLE()),
-						*object.NewAttribute("timestamp", strconv.FormatUint(blk.Timestamp, 10)),
+						*object.NewAttribute("Primary", strconv.Itoa(int(blk.PrimaryIndex))),
+						*object.NewAttribute("Hash", blk.Hash().StringLE()),
+						*object.NewAttribute("PrevHash", blk.PrevHash.StringLE()),
+						*object.NewAttribute("Timestamp", strconv.FormatUint(blk.Timestamp, 10)),
+						*object.NewAttribute("Magic", magic),
```
Comment (roman-khimov) [inline cli/util/uploader.go:170]: I thought we want this per-container, not per-object. Or you want to store multiple chains in a single container?

File: cli/util/uploader.go [hunk 4]
```diff
@@ -394,7 +400,8 @@ func updateIndexFiles(ctx *cli.Context, p *pool.Pool, containerID cid.ID, accoun
 		}
 		attrs := []object.Attribute{
 			*object.NewAttribute(attributeKey, strconv.Itoa(int(i))),
-			*object.NewAttribute("size", strconv.Itoa(int(indexFileSize))),
+			*object.NewAttribute("Size", strconv.Itoa(int(indexFileSize))),
```
Comment (roman-khimov) [inline cli/util/uploader.go:403]: Better some `IndexSize` or `IndexLength`, `Size` attribute is too easy to confuse with object size.

File: cli/util/uploader.go [hunk 5]
```diff
@@ -109,6 +103,27 @@ func uploadBin(ctx *cli.Context) error {
 		return cli.Exit(fmt.Errorf("failed to get network info: %w", err), 1)
 	}
 	homomorphicHashingDisabled := net.HomomorphicHashingDisabled()
+	container, err := p.ContainerGet(ctx.Context, containerID, client.PrmContainerGet{})
```
Comment (AnnaShaleva) [inline cli/util/uploader.go:106]: I'm starting to be extremely suspicious regarding all operations with pool. Can we use retry here?

**Review by AnnaShaleva — COMMENTED** [review]
Otherwise LGTM.

File: cli/util/uploader.go (hunk 5 above)
Comment (AliceInHunterland) [reply cli/util/uploader.go:106]: also we can add retry here: `net, err := p.NetworkInfo(ctx.Context, client.PrmNetworkInfo{})` will do it in https://github.com/nspcc-dev/neo-go/pull/3638#discussion_r1812175065

**Review by AnnaShaleva — DISMISSED** [review]

**Comment by AnnaShaleva** [comment]
```
2024-10-23T08:34:53.2882722Z failed to load account: open wallet: open ./not-exist.json: no such file or directory
2024-10-23T08:34:53.2883695Z failed to decode container ID: invalid length 3
2024-10-23T08:34:53.2885040Z failed to create RPC client: failed to get network magic: Post "https://test": dial tcp: lookup test on 127.0.0.53:53: server misbehaving
2024-10-23T08:34:53.2887105Z     logger.go:146: 2024-10-23T08:34:47.157Z	DEBUG	processing rpc request	{"method": "getversion", "params": "[]"}
2024-10-23T08:34:53.2889593Z     logger.go:146: 2024-10-23T08:34:47.157Z	DEBUG	processing rpc request	{"method": "getnativecontracts", "params": "[]"}
2024-10-23T08:34:53.2892126Z     logger.go:146: 2024-10-23T08:34:47.242Z	DEBUG	done processing headers	{"headerIndex": 2, "blockHeight": 1, "took": "167.934µs"}
2024-10-23T08:34:53.2894727Z     logger.go:146: 2024-10-23T08:34:47.343Z	DEBUG	done processing headers	{"headerIndex": 3, "blockHeight": 2, "took": "175.388µs"}
2024-10-23T08:34:53.2896538Z failed to dial NeoFS pool: at least one node must be healthy
2024-10-23T08:34:53.2897437Z     executor.go:239: 
2024-10-23T08:34:53.2898879Z         	Error Trace:	/home/runner/work/neo-go/neo-go/internal/testcli/executor.go:239
2024-10-23T08:34:53.2901236Z         	            				/home/runner/work/neo-go/neo-go/internal/testcli/executor.go:244
2024-10-23T08:34:53.2903266Z         	            				/home/runner/work/neo-go/neo-go/cli/util/util_test.go:212
2024-10-23T08:34:53.2904473Z         	Error:      	Received unexpected error:
2024-10-23T08:34:53.2905309Z         	            	EOF
2024-10-23T08:34:53.2905991Z         	Test:       	TestUploadBin
2024-10-23T08:34:53.2907547Z     logger.go:146: 2024-10-23T08:34:47.380Z	INFO	shutting down RPC server	{"endpoint": "127.0.0.1:40959"}
2024-10-23T08:34:53.2910442Z     logger.go:146: 2024-10-23T08:34:47.380Z	INFO	persisted to disk	{"blocks": 3, "keys": 163, "headerHeight": 3, "blockHeight": 3, "took": "68.588µs"}
2024-10-23T08:34:53.2912386Z --- FAIL: TestUploadBin (0.24s)
2024-10-23T08:34:53.2912861Z FAIL
```

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**